integration/ledger/snapshot_test.go

```diff
@@ -0,0 +1,816 @@
+		fileContent1, err := ioutil.ReadFile(filepath.Join(snapshotDir1, file))
+		Expect(err).NotTo(HaveOccurred())
+		fileContent2, err := ioutil.ReadFile(filepath.Join(snapshotDir2, file))
+		Expect(err).NotTo(HaveOccurred())
+		Expect(fileContent1).To(Equal(fileContent2))
+	}
+}
+
+func startPeerInExistingOrgFromSnapshot(s *setup, orgName, peerName, channelID, snapshotDir string, lastBlockInSnapshot int, useCouchDB bool) (*nwo.Peer, ifrit.Process) {
+	peer := &nwo.Peer{
+		Name:         peerName,
+		Organization: orgName,
+		Channels: []*nwo.PeerChannel{
+			{Name: channelID},
+		},
+	}
+	s.network.Peers = append(s.network.Peers, peer)
+	s.peers = append(s.peers, peer)
+
+	var couchProc ifrit.Process
+	if useCouchDB {
+		By("starting couch process and configuring it for peer " + peer.ID())
+		couchProc = configPeerWithCouchDB(s, peer)
+	}
+
+	By("starting the new peer " + peer.ID())
+	s.startPeer(peer)
+
+	return peer, couchProc
+}
+
+func startPeerInNewOrgFromSnapshot(s *setup, org *nwo.Organization, newPeerName, channelID, snapshotDir string, lastBlockInSnapshot int, useCouchDB bool) (*nwo.Peer, ifrit.Process) {
+	peer := &nwo.Peer{
+		Name:         newPeerName,
+		Organization: org.Name,
+		Channels:     []*nwo.PeerChannel{{Name: channelID, Anchor: true}},
+	}
+	s.network.AddOrg(org, peer)
+	generateOrgUpdateMaterials(s.network, peer)
+	s.peers = append(s.peers, peer)
+
+	addNewOrgToChannel(s.network, s.orderer, org, channelID)
+
+	var couchProc ifrit.Process
+	if useCouchDB {
+		By("starting couch process and configuring it for peer " + peer.ID())
+		couchProc = configPeerWithCouchDB(s, peer)
+	}
+
+	By("starting the new peer " + peer.ID())
+	s.startPeer(peer)
+
+	return peer, couchProc
+}
+
+func addNewOrgToChannel(n *nwo.Network, orderer *nwo.Orderer, org *nwo.Organization, channelID string) {
+	By("getting current channel config from an existing peer")
+	existingPeers := n.PeersWithChannel(channelID)
+	Expect(len(existingPeers) > 0).Should(BeTrue())
+	currentConfig := nwo.GetConfig(n, existingPeers[0], orderer, channelID)
+	updatedConfig := proto.Clone(currentConfig).(*common.Config)
+
+	By("updating the channel config to include a new org")
+	sess, err := n.ConfigTxGen(commands.PrintOrg{
+		ConfigPath: n.RootDir,
+		PrintOrg:   org.Name,
+	})
+	Expect(err).NotTo(HaveOccurred())
+	Eventually(sess, n.EventuallyTimeout).Should(gexec.Exit(0))
+	orgGroup := &ordererext.DynamicOrdererOrgGroup{ConfigGroup: &common.ConfigGroup{}}
+	err = protolator.DeepUnmarshalJSON(bytes.NewBuffer(sess.Out.Contents()), orgGroup)
+	Expect(err).NotTo(HaveOccurred())
+	updatedConfig.ChannelGroup.Groups["Application"].Groups[org.Name] = orgGroup.ConfigGroup
+	nwo.UpdateConfig(n, orderer, channelID, currentConfig, updatedConfig, true, existingPeers[0], n.Peers...)
+}
+
+func joinBySnapshot(n *nwo.Network, orderer *nwo.Orderer, peer *nwo.Peer, channelID string, snapshotDir string, lastBlockInSnapshot int) {
```

Comment:
`lastBlockInSnapshot` can be avoided as inside the function the `channelHeight` is computed explicitly.

Reply:
We need `lastBlockInSnapshot` to verify the bootstrapping snapshot info. `channelHeight` is for current channel height only.